File: plugins/capawesome-capacitor-bluetooth-low-energy/android/src/main/java/io/capawesome/capacitorjs/plugins/bluetoothle/classes/options/WriteDescriptorOptions.java
package io.capawesome.capacitorjs.plugins.bluetoothle.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import io.capawesome.capacitorjs.plugins.bluetoothle.BluetoothLowEnergyHelper;
import org.json.JSONException;

public class WriteDescriptorOptions {

    @NonNull
    private String characteristicId;

    @NonNull
    private String descriptorId;

    @NonNull
    private String deviceId;

    @NonNull
    private String serviceId;

    @NonNull
    private byte[] value;

    public WriteDescriptorOptions(
        @NonNull String characteristicId,
        @NonNull String descriptorId,
        @NonNull String deviceId,
        @NonNull String serviceId,
        @NonNull JSArray value
    ) throws JSONException {
        this.characteristicId = characteristicId;
        this.descriptorId = descriptorId;
        this.deviceId = deviceId;
        this.serviceId = serviceId;
        this.value = BluetoothLowEnergyHelper.convertJsonArrayToByteArray(value);
    }

    @NonNull
    public String getCharacteristicId() {
        return characteristicId;
    }

    @NonNull
    public String getDescriptorId() {
        return descriptorId;
    }

    @NonNull
    public String getDeviceId() {
        return deviceId;
    }

    @NonNull
    public String getServiceId() {
        return serviceId;
    }

    public byte[] getValue() {
        return value;
    }
}

File: plugins/capawesome-capacitor-bluetooth-low-energy/ios/Plugin/Classes/Options/WriteDescriptorOptions.swift
import Foundation
import Capacitor

@objc public class WriteDescriptorOptions: NSObject {
    private var characteristicId: String
    private var descriptorId: String
    private var deviceId: String
    private var serviceId: String
    private var value: Data

    init(characteristicId: String, descriptorId: String, deviceId: String, serviceId: String, value: [UInt8]) {
        self.characteristicId = characteristicId
        self.descriptorId = descriptorId
        self.deviceId = deviceId
        self.serviceId = serviceId
        self.value = BluetoothLowEnergyHelper.convertJsonArrayToData(values: value)
    }

    func getCharacteristicId() -> String {
        return characteristicId
    }

    func getDescriptorId() -> String {
        return descriptorId
    }

    func getDeviceId() -> String {
        return deviceId
    }

    func getServiceId() -> String {
        return serviceId
    }

    func getValue() -> Data {
        return value
    }
}
